### PR TITLE
Issue #7879: Missing JDBC TIMESTAMP_TZ

### PR DESCRIPTION
**Pull request by hawkfish**
Add some cases where converting TIMESTAMP_TZ is generating unparseable strings instead of just doing what TIMESTAMP does.

Unsure on how to test this, so suggestions from Java nerds welcome! It did fix a problem I found in Tableau.

### CODE REVIEW

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java [hunk 1]
```diff
@@ -533,12 +533,9 @@ public static void test_throw_wrong_datatype() throws Exception {
...
 		rs.next();
-
-		try {
-			rs.getTimestamp(2);
-			fail();
-		} catch (IllegalArgumentException e) {
-		}
+		
+		//	This succeeds now,
+		rs.getTimestamp(2);
```
Comment (Mause) [inline tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:538]: We should probably keep this test as one that tests for the exception? So maybe switch to another data type (bool maybe?)

File: tools/jdbc/src/main/java/org/duckdb/DuckDBResultSet.java [hunk 2]
```diff
@@ -1194,7 +1194,8 @@ public <T> T getObject(int columnIndex, Class<T> type) throws SQLException {
 			}
 		} else if (type == Long.class) {
 			if (sqlType == DuckDBColumnType.BIGINT
-					|| sqlType == DuckDBColumnType.TIMESTAMP) {
+					|| sqlType == DuckDBColumnType.TIMESTAMP
```
Comment (Mause) [inline tools/jdbc/src/main/java/org/duckdb/DuckDBResultSet.java:1197]: Might be worth pulling out these sets of comparisons? `isTimestampType`, `isLongType`, etc

**Review by Mause — CHANGES_REQUESTED** [review]
> Unsure on how to test this, so suggestions from Java nerds welcome!

Something like this maybe?

```java
Statement rs = statement.executeQuery("SELECT now()::TIMEZONE_TZ");
rs.next();
rs.getObject(1, Long.class);
```

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**